src/INTERLAYER/pair_ilp_graphene_hbn.cpp

```diff
@@ -56,9 +57,15 @@ static const char cite_ilp[] =
     " year =    2018,\n"
     "}\n\n";
 
+// to indicate which potential style was used in outputs
+static std::map<int, std::string> variant_map = {
```

Comment:
not clear why this should be a static variable?

Reply:
what I mean is, is there a better way to do this w/out a static variable?

Reply:
because it is relevant for and should be visible only in this particular file and this has less overhead than adding it to the class as a member object.
This way the class can use the map to customize error messages to print the correct style name when there are errors from derived classes.

Reply:
ok - makes sense

Reply:
> what I mean is, is there a better way to do this w/out a static variable?

I don't get your point. IMO a static variable (mind you this is actually a class instance) is the best solution that I can think of.
Why is 'static' bad in this case? Dropping the static would make it globally visible and then it would need to declared in a suitable namespace to avoid conflicts.